docs/notes.md

```diff
@@ -603,4 +603,242 @@ sends a pair \\((L\_j,R\_j)\\) of points at each step \\(j = k\dots1\\). An
 additional and final step involves sending a pair of scalars
 \\((a^{(0)}\_0,b^{(0)}\_0)\\) and checking the final relation directly.
 
-[bulletproofs_paper]: https://eprint.iacr.org/2017/1066.pdf
+Aggregated Range Proof
+======================
+
+We want to create an aggregated range proof for `m` values (`m` parties) that is more efficient to create and verify than `m` individual range proofs.
+
+The aggregated range proof has the same form as the individual range proof, but differs in that different parties are seperated by different powers of the challenge scalars `y` and `z`.
+
+We will explain how one piece of the aggregated proof is generated for party `j`, and then will show how all of the pieces for all of the `m` parties can be combined into one aggregated proof.
+
+Party `j` begins with a secret value \\(v_j\\), and wishes to convince the verifier that \\(v_j \in [0, 2^n)\\) without revealing \\(v_j\\).
+
+Proving range statements with bit vectors
+-----------------------------------------
+
+Let \\({\mathbf{a}}\_{Lj}\\) be the vector of bits of \\(v_j\\).
+Then \\(v_j\\) can be represented as:
+\\[
+\begin{aligned}
+  v_j &= {\langle {\mathbf{a}}\_{Lj}, {\mathbf{2}}^{n} \rangle}
+\end{aligned}
+\\]
+We need \\({\mathbf{a}}\_{Lj}\\) to be a vector of integers \\(\\{0,1\\}\\).
```

Comment:
Notation: I think `_{Lj}` is confusing. I'm not 100% sure of the notationally-best way to write the party index.

Superscripts would have to have `(j)` or something to distinguish them from exponents, but we're already overloading exponents with the exponential-vector thing.

Subscripts don't have the same confusion, but we're already using subscripts. Maybe the least awkward thing would be to use comma separated subscripts, like `a_{L, j}`?

Reply:
makes sense - I was looking for a better way to do this, and `a_{L, j}` is better than the current way.